pkg/export/setup/setup_test.go

```diff
@@ -35,19 +35,34 @@ func TestFromFlags_NotOnGCE(t *testing.T) {
 		t.Skip("This test can't run on GCP or Cloudtop; we expect no metadata server.")
 	}
 
+	logger := log.NewLogfmtLogger(os.Stderr)
 	fake := kingpin.New("test", "test")
-	newExport := FromFlags(fake, "product")
-	// Our readines is default (3 * 10s), so ensure FromFlags is never longer than 30s.
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
+
+	exporterOpts := export.ExporterOpts{}
+	exporterOpts.Default()
+	ExporterOptsFlags(fake, &exporterOpts)
+
+	metadataOpts := MetadataOpts{}
+	metadataOpts.Default()
+	metadataOpts.SetupFlags(fake)
 
 	// Fake app invocation.
 	if _, err := fake.Parse(nil); err != nil {
 		t.Fatal(err)
 	}
 
-	// Make sure constructor does is not stuck forever.
-	_, _ = newExport(ctx, log.NewLogfmtLogger(os.Stderr), nil)
+	done := make(chan struct{}, 1)
+	go func() {
+		metadataOpts.ExtractMetadata(logger, &exporterOpts)
```

Comment:
> @pintohutch [`7fe3d0b` (#974)](https://github.com/GoogleCloudPlatform/prometheus-engine/pull/974/commits/7fe3d0b10a0a0258d55440ef27bc473102e02fe1#r1661217052)
> question: If we do want to make this change, wouldn't we also need to update the tests in `TestTryPopulateUnspecifiedFromMetadata` to ensure timeouts are correctly updated and respected in the config?

Can you clarify why we might need to? The default/logic hasn't been changed, only moved.